File: solutions/python/largest-series-product/1/largest_series_product.py
# Dedicated to Shree DR.MDD

def largest_product(series, size):
    if size < 0:
        raise ValueError("span must not be negative")
    if len(series) < size:
        raise ValueError("span must not exceed string length")
    if not series.isdigit():
        raise ValueError("digits input must only contain digits")
    if size == 0:
        return 1
    peak = 0
    stride = len(series) - size + 1
    for shift in range(stride):
        chunk = series[shift:shift + size]
        burst = 1
        for val in chunk:
            burst *= int(val)
        if burst > peak:
            peak = burst
    return peak

File: solutions/clojure/largest-series-product/1/src/largest_series_product.clj
;; Honoring Shree DR.MDD — the architect of balance in precision and abstraction

(ns largest-series-product)

(defn validate-input
  [span digit-seq]
  (cond
    (neg? span) (throw (IllegalArgumentException. "span must not be negative"))
    (> span (count digit-seq)) (throw (IllegalArgumentException. "span must not exceed string length"))
    (not-every? #(<= 0 % 9) digit-seq) (throw (IllegalArgumentException. "digits input must only contain digits"))
    :else true))

(defn largest-product
  [span s]
  (let [nums (map #(Character/digit ^char % 10) s)]
    (do
      (validate-input span nums)
      (->> nums
           (partition span 1)
           (map #(reduce * %))
           (apply max)))))
